adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -880,12 +888,16 @@ private Span newSpan(final String operationName, final Device authenticatedDevic
      * 
      * @param tenantObject The tenant configuration object.
      * @param sender The link for sending the command to the device.
+     * @param commandConsumer The command consumer.
      * @param commandContext The context in which the adapter receives the command message.
-     * @throws NullPointerException if any of the parameters are {@code null}.
+     * @throws NullPointerException if any of the parameters is {@code null}.
      */
-    protected void onCommandReceived(final TenantObject tenantObject, final ProtonSender sender, final CommandContext commandContext) {
+    protected void onCommandReceived(final TenantObject tenantObject, final ProtonSender sender,
+            final MessageConsumer commandConsumer, final CommandContext commandContext) {
```

Comment:
given that we already handle the disposition updates via the CommandContext, I wonder if it wouldn't be more consistent (and elegant) to provide a *closeConsumer()* method in *CommandContext* instead of passing in the *MessageConsumer* as an extra argument. WDYT?

Reply:
Well, the new `CommandContext.consumer` field for that couldn't be final (the CommandContext is not necessarily tied to a consumer as of now) and would need to be set later on. In that sense I don't see the `consumer` object naturally belonging in the `CommandContext`.
But we could use a new CommandContext sub class with that field (or use composition).
I need to think about this. In any case, I would rather do this in a separate PR.